src/coreComponents/python/modules/geosx_xml_tools_package/geosx_xml_tools/main.py

```diff
@@ -0,0 +1,128 @@
+"""Command line tools for geosx_xml_tools"""
+
+import sys
+import argparse
+import os
+import time
+from geosx_xml_tools import xml_processor
+
+
+def parse_arguments():
+    # Parse the user arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-i', '--input', type=str, action='append', help='Input file name (multiple allowed)')
+    parser.add_argument('-c', '--compiled-name', type=str, help='Compiled xml file name (otherwise, it is randomly genrated)', default='')
+    parser.add_argument('-s', '--schema', type=str, help='GEOSX schema to use for validation', default='')
+    parser.add_argument('-v', '--verbose', type=int, help='Verbosity of outputs', default=0)
+    parser.add_argument('-p', '--parameters', nargs='+', action='append', help='Parameter overrides (name value, multiple allowed)', default=[])
+    return parser.parse_known_args()
+
+
+def check_mpi_rank():
+    rank = 0
+    mpi_rank_key_options = ['OMPI_COMM_WORLD_RANK', 'PMI_RANK']
+    for k in mpi_rank_key_options:
+        if k in os.environ:
+            rank = int(os.environ[k])
+    return rank
+
+
+def wait_for_file_write_rank_0(target_file_argument=0, max_wait_time=100):
+    def wait_for_file_write_rank_0_inner(writer):
+        def wait_for_file_write_rank_0_decorator(*args, **kwargs):
+            # Check the target file status
+            rank = check_mpi_rank()
+            fname = ''
+            if isinstance(target_file_argument, int):
+                fname = args[target_file_argument]
+            else:
+                fname = kwargs[target_file_argument]
+
+            target_file_exists = os.path.isfile(fname)
+            target_file_edit_time = 0
+            if target_file_exists:
+                target_file_edit_time = os.path.getmtime(fname)
+
+            # Go into the target process or wait for the expected file update
+            if (rank == 0):
+                return writer(*args, **kwargs)
+            else:
+                ta = time.time()
+                while(time.time() - ta < max_wait_time):
+                    if target_file_exists:
+                        if (os.path.getmtime(fname) > target_file_edit_time):
+                            break
+                    else:
+                        if os.path.isfile(fname):
+                            break
+                    time.sleep(0.1)
+                print('File detected')
+        return wait_for_file_write_rank_0_decorator
+    return wait_for_file_write_rank_0_inner
```

Comment:
@klevzoff - With some experimentation, I think I found a decent backup option.  This decorator function will check for the current rank by looking at environment variables, attempt to do the preprocessing on rank 0, and wait for the file to be created/updated on the remaining processes.  At worst, this should fall back to the current method, which produces (seemingly benign) file warnings in python.

Reply:
Hey @cssherman, I'm testing this on my local machine, and with 4 or more ranks I occasionally have the job stall for the entire timeout value (100 s). I think this might happen when rank 0 manages to complete writing the file *before* one of the other ranks sets `target_file_edit_time` - in that case the modification time stamp doesn't ever change anymore, and that rank is stuck waiting.

Reply:
Interesting.  I think that the appropriate thing here is to add a bit of room for the time check.  This would add an implicit maximum frequency for writing these files, but I don't think that would be an issue.  Perhaps something like this:

`target_file_edit_time = os.path.getmtime(fname) - 10`